rcsb/utils/dictionary/DictMethodCompModelHelper.py

```diff
@@ -166,28 +167,30 @@ def addPolymerEntityTaxonomy(self, dataContainer, catName, **kwargs):
             if not all([ai in dataContainer.getObj('ma_target_ref_db_details').getAttributeList() for ai in ["ncbi_taxonomy_id", "organism_scientific"]]):
                 return False
             geneName = None
-            if dataContainer.exists("af_target_ref_db_details"):
-                tObj = dataContainer.getObj("af_target_ref_db_details")
-                geneName = tObj.getValue("gene", 0)
 
             epLenD = self.__commonU.getPolymerEntityLengths(dataContainer)
             tObj = dataContainer.getObj("ma_target_ref_db_details")
             if not dataContainer.exists("entity_src_nat"):
                 dataContainer.append(DataCategory("entity_src_nat", attributeNameList=atL))
             sObj = dataContainer.getObj("entity_src_nat")
             #
+            jj = 0
             for ii in range(tObj.getRowCount()):
                 taxId = tObj.getValue("ncbi_taxonomy_id", ii)
                 orgName = tObj.getValue("organism_scientific", ii)
                 entityId = tObj.getValue("target_entity_id", ii)
+                geneName = tObj.getValueOrDefault("gene_name", ii, defaultValue=None)
+                dbName = tObj.getValue("db_name", ii)
                 #
-                sObj.setValue(entityId, "entity_id", ii)
-                sObj.setValue(taxId, "pdbx_ncbi_taxonomy_id", ii)
-                sObj.setValue(orgName, "pdbx_organism_scientific", ii)
-                sObj.setValue("1", "pdbx_src_id", ii)
-                sObj.setValue("1", "pdbx_beg_seq_num", ii)
-                sObj.setValue(str(epLenD[entityId]), "pdbx_end_seq_num", ii)
-                sObj.setValue(geneName, "rcsb_gene_name", ii)
+                if dbName == "UNP":
+                    sObj.setValue(entityId, "entity_id", jj)
+                    sObj.setValue(taxId, "pdbx_ncbi_taxonomy_id", jj)
+                    sObj.setValue(orgName, "pdbx_organism_scientific", jj)
+                    sObj.setValue("1", "pdbx_src_id", jj)
+                    sObj.setValue("1", "pdbx_beg_seq_num", jj)
+                    sObj.setValue(str(epLenD[entityId]), "pdbx_end_seq_num", jj)
+                    sObj.setValue(geneName, "rcsb_gene_name", jj)
+                    jj += 1
```

Comment:
How would this handle the case where the UNP row comes _after_ the OrthoDB row in `ma_target_ref_db_details`? If this loop is incrementing `jj`, and is using that value for the `sObj.setValue()` calls, then wouldn't it end up leaving a gap in row assignments? Or does that not matter?

Reply:
Oh wait, never mind, my bad. I see now, this should handle that case just fine since it only increments `jj` if `dbName == "UNP"`. Please disregard my question.